code/game/objects/items/cards_ids.dm

```diff
@@ -265,23 +268,41 @@
 /obj/item/card/id/GetID()
 	return src
 
+/obj/item/card/id/update_icon(blank=FALSE)
+	cut_overlays()
+	if(!uses_overlays)
+		return
+	var/job = assignment ? ckey(GetJobName()) : null
+	var/list/add_overlays = list()
+	if(!blank)
+		add_overlays += mutable_appearance(icon, "assigned")
+	if(job)
+		add_overlays += mutable_appearance(icon, "id[job]")
+	add_overlay(add_overlays)
+	if(istype(loc, /obj/item/storage/wallet))
+		var/obj/item/storage/wallet/powergaming = loc
+		if(powergaming.front_id == src)
+			powergaming.update_icon(add_overlays)
+
+/obj/item/card/id/get_examine_string(mob/user, thats = FALSE)
+	if(uses_overlays)
+		return "[costly_icon2html(src, user)] [thats? "That's ":""][get_examine_name(user)]" //displays all overlays in chat

```

Comment:
cache the result of costly_icon2html, it calls getflaticon which is very expensive